sstables/mx/reader.hh

```diff
@@ -50,7 +50,8 @@ mutation_reader make_full_scan_reader(
         reader_permit permit,
         tracing::trace_state_ptr trace_state,
         read_monitor& monitor,
-        sstable::integrity_check integrity);
+        sstable::integrity_check integrity,
+        digest_validation_result* digest_result);
```

Comment:
Like I already said, I think this output param is not needed. Instead, we should add digest check request into `integrity_check` flag.
The caller of this method, can check itself whether digest component exists or not, so it will know if digest checking is possible.
All errors are already signaled via exceptions, so caller will know if digest checking failed and why.

Reply:
Let me explain why I thought an output parameter would be a better option than an exception:
1. A valid result is not explicitly communicated to the user. They have to figure it out from the context. Digest check spans across many `get()` calls, so after a non-throwing `get()` it could be that either digest is valid, or calculation is still in progress. I think it's better to provide this information explicitly.
2. In case of an exception, there is no way to tell whether it's about checksums or digest. I think it would be nice to extend `validate_checksums()` in a next PR to expose this information.
3. The code is more robust against early closes. If at some point the caller "thinks" they have read the whole file, but the result is `in_progress`, it means there is a bug somewhere.